src/raftstore/store/worker/pd.rs

```diff
@@ -157,7 +165,64 @@ impl<T: PdClient> Runner<T> {
         handle.spawn(f);
     }
 
-    fn handle_store_heartbeat(&self, handle: &Handle, stats: pdpb::StoreStats) {
+    fn handle_store_heartbeat(&self,
+                              handle: &Handle,
+                              mut stats: pdpb::StoreStats,
+                              store_info: StoreInfo) {
+        let disk_stats = match fs2::statvfs(store_info.engine.path()) {
+            Err(e) => {
+                error!("get disk stat for rocksdb {} failed: {}",
+                       store_info.engine.path(),
+                       e);
+                return;
+            }
+            Ok(stats) => stats,
+        };
+
+        let disk_cap = disk_stats.total_space();
+        let capacity = if store_info.capacity == 0 || disk_cap < store_info.capacity {
+            disk_cap
+        } else {
+            store_info.capacity
+        };
+        stats.set_capacity(capacity);
+
+        let mut used_size = flush_engine_properties_and_get_used_size(store_info.engine.clone());
```

Comment:
I prefer using get_used_size directly, you should flush engine properties to the flush thread.